### ISSUE TITLE
Super danganronpa 2 interacting with characters problem

### ISSUE THREAD

**Issue by Ala7x** [hrydgard/ppsspp#13321]
-Game:Super danganronpa 2
-PPSSPP:1.10.3 (tried other builds too)
-Didn't work before but danganronpa 1 did work for me when I played it 2 years ago
-the issues is that I can't interact with characters most of the time (except for the white haired guy /I believe I can interact with others when its on 3d roaming)
The crosshair can detect object but not characters (at least where I reached in the game)
-tried everything (changing setting and old build yet it doesn't work)
-I'm using Leagoo Z6 with android 6.0 marshmallow 
-I'm using openGL
![Screenshot_20200822-155619](https://user-images.githubusercontent.com/70069383/90961256-7e698900-e49f-11ea-86b7-e29aa66bf5d5.png)

**Comment by Gamemulatorer** [comment]
You better play this game using android port of this game that is release today :)
https://play.google.com/store/apps/details?id=jp.co.spike_chunsoft.DR2&referrer=utm_source%3Dgoogle%26utm_medium%3Dorganic%26utm_term%3Ddangan+ronpa+2+playstore

**Comment by Ala7x** [comment]
OK now its legal to commit suicide 
![Screenshot_20200824-174536](https://user-images.githubusercontent.com/70069383/91072602-aa1c7880-e631-11ea-8653-9bfb347cc784.png)

**Comment by Panderner** [comment]
> OK now its legal to commit suicide
> ![Screenshot_20200824-174536](https://user-images.githubusercontent.com/70069383/91072602-aa1c7880-e631-11ea-8653-9bfb347cc784.png)

@Ala7x Also official Danganronpa games on Google Play Store isn't compatible for your phone due to your phone is below Android 7.0

**Comment by Ala7x** [comment]
Yeah I know damn college draining my money or else I will buy a new phone 
Anyway any news to fix the issue ?

**Comment by Gamemulatorer** [comment]
Better to update your phone os via official update or custom rom to play the android port..

this game has the same bug on my phone w/ mali-450 gpu and unfortunately this issue can't be fixed in the near future 🤷

**Comment by Ala7x** [comment]
What is the custom ROM tho?

**Comment by Panderner** [comment]
@Ala7x maybe you used skip buffered rendering

**Comment by Panderner** [comment]
Danganronpa 2 Requires Buffered Rendering in order to interact objects and characters.

Skip Buffer Effects:
![Screenshot_2020-12-07-00-47-36-61_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101286568-23795900-3826-11eb-89e6-00ca8ddd8226.png)

Buffered Rendering:
![Screenshot_2020-12-07-00-47-44-58_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101286576-368c2900-3826-11eb-9fba-f6785457d185.png)

**Comment by Panderner** [comment]
@Ala7x have you tried to use Buffered Rendering?

**Comment by Panderner** [comment]
@Ala7x no feedback?

**Comment by Panderner** [comment]
Buffered Rendering in Danganronpa fixes Collision detection related bugs. but it's slow for in-game outside of Talking sequences and class trials and incorrect FPS counter.

Buffered Rendering:
![Screenshot_2020-12-08-20-11-27-41_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101482774-242efe00-3992-11eb-9da9-a31fa9eb1ae3.png)
![Screenshot_2020-12-08-20-10-14-80_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101482837-3f017280-3992-11eb-8bac-82b1298b9335.png)
![Screenshot_2020-12-08-20-06-24-84_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101482921-61938b80-3992-11eb-89d4-8075f725f501.png)

Skip Buffer Effects:
![Screenshot_2020-12-08-20-11-36-46_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101483089-9e5f8280-3992-11eb-8e67-642aeff2fd5a.png)
![Screenshot_2020-12-08-20-08-58-54_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101483146-b59e7000-3992-11eb-9216-cd4fa78644be.png)
![Screenshot_2020-12-08-20-06-17-00_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/101483216-cd75f400-3992-11eb-9544-34b1c128ee05.png)

**Comment by hrydgard** [comment]
Buffered rendering is simply required for this game to function, I don't think there's much to do.

**Comment by Ala7x** [comment]
Sry for not closing the topic but i didn't get solution and waited until i forgot about it 
i just got a pc and played it there

**Comment by hrydgard** [comment]
I'm gonna close this until we get more reports.

**Comment by Panderner** [comment]
@Ala7x have you used buffered rendering for your android phone?

**Comment by Ala7x** [comment]
> @Ala7x have you used buffered rendering for your android phone?

yeah i did but it didn't work sadly

**Comment by Panderner** [comment]
> > @Ala7x have you used buffered rendering for your android phone?
> 
> yeah i did but it didn't work sadly

Maybe you used skip buffer effects.

**Comment by Ala7x** [comment]
> > > @Ala7x have you used buffered rendering for your android phone?
> > 
> > 
> > yeah i did but it didn't work sadly
> 
> Maybe you used skip buffer effects.

i used every thing man i even downloaded the old version of ppsspp  so i can try "read to memory (cpu) /read to memory (gpu) " even the 0.9 builds but all for nothing

**Comment by hrydgard** [comment]
Using any special "game optimization" settings or something on the device? I don't understand why this wouldn't work..

**Comment by Panderner** [comment]
I said My Realme C2 and POCO M3 works fine for me:
![Screenshot_2020-12-23-11-38-27-533_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/102956943-1f328880-4514-11eb-8ea3-4ed0fe949f7b.jpg)

**Comment by Panderner** [comment]
@Ala7x maybe your phone has driver bugs for this game

**Comment by Ala7x** [comment]
I'm not sure .didn't used any game optimization either it's just didn't work no matter what 
I literally tried every option even the ones that I'm sure will not change anything
But I'm 100% sure that I played Danganronpa 1 with no problems in the same device 
And btw Mana khemilia doesn't work on that device it crashes right at the loading on that device maybe the problems are realted ?

**Comment by Panderner** [comment]
> Mana khemilia doesn't work on that device it crashes right at the loading on that device maybe the problems are realted ?

Mana Khemia? It runs fine and doesn't crash for me.
@Ala7x maybe your iso is corrupted

**Comment by Ala7x** [comment]
> > Mana khemilia doesn't work on that device it crashes right at the loading on that device maybe the problems are realted ?
> 
> Mana Khemia? It runs fine and doesn't crash for me.
> @Ala7x maybe your iso is corrupted
Nah I'm pretty sure the iso is fine 
I mean i even tried a different one but still I'm not sure why it didn't work 
I guess we are getting side tracked here but if it really is related then we might be on something

**Comment by yato442** [comment]
@Ala7x 
Hi,guys!
The same problem for me and I can't pass the 2nd part in ppspp because of I cant interact with Peko Pekoyama.
Any solution to the problem and can I know what ppspp version and iso of the game.
Ty if u can help me,I want to play it so badly.

**Comment by Gamemulatorer** [comment]
> @Ala7x Hi,guys! The same problem for me and I can't pass the 2nd part in ppspp because of I cant interact with Peko Pekoyama. Any solution to the problem and can I know what ppspp version and iso of the game. Ty if u can help me,I want to play it so badly.

Save state of the game near the issue please upload I will try to reproduce.

**Comment by yato442** [comment]
I already deleted my save,can u give me a savestate instead after the intro part where I need to talk with Peko Pekoyama. I want to play it so badly. Thank you if u can grant my request.

**Comment by Panderner** [comment]
> https://gamefaqs.gamespot.com/psp/604654-dangan-ronpa/saves

Nope No Danganronpa 2 save files here for gamefaqs. Only for Danganronpa 1

**Comment by yato442** [comment]
> > https://gamefaqs.gamespot.com/psp/604654-dangan-ronpa/saves
> 
> Nope No Danganronpa 2 save files here for gamefaqs. Only for Danganronpa 1
Did you have a personal save? Just want to play it. Just past the part that you need to interact with them one by one. Anyone that have the save. I appreciate it and thank you in advance.

**Comment by unknownbrackets** [comment]
In case helpful, this should be a 100% complete save, courtesy of a minute of googling:

[NPJH50631DATKG0000.zip](https://github.com/hrydgard/ppsspp/files/7358840/NPJH50631DATKG0000.zip)

-[Unknown]

**Comment by yato442** [comment]
> In case helpful, this should be a 100% complete save, courtesy of a minute of googling:
> 
> [NPJH50631DATKG0000.zip](https://github.com/hrydgard/ppsspp/files/7358840/NPJH50631DATKG0000.zip)
> 
> -[Unknown]

Thank you very much. I really appreciated it. I can play the game now ty to you and whoever owns the savedata,thank you also for uploading it.

**Comment by Gamemulatorer** [comment]
> > In case helpful, this should be a 100% complete save, courtesy of a minute of googling:
> > [NPJH50631DATKG0000.zip](https://github.com/hrydgard/ppsspp/files/7358840/NPJH50631DATKG0000.zip)
> > -[Unknown]
> 
> Thank you very much. I really appreciated it. I can play the game now ty to you and whoever owns the savedata,thank you also for uploading it.

Did you reproduce the issue using ppsspp v1.12.2?

**Comment by Gamemulatorer** [comment]
@Ala7x no feedback outdated?

**Comment by Gamemulatorer** [comment]
can you still reproduce this issue in v1.14?

**Comment by ghvxx43696** [comment]
Well the issue isn't outdated nor platform specific since this bug affects version 1.19.3 on iOS.

This bug is related to rendering backend and texture scaling. I'm not sure about the rendering resolution. I didn't get clear results here, although for a moment I had the impression that selecting a character with the crosshairs is more difficult at resolutions higher than 1x PSP.

My observations: interaction with characters is possible on OpenGL. On Vulkan, selecting characters for interaction is difficult. Sometimes impossible (if they are standing a little further away), sometimes possible by “aiming” at the sprite's legs. For both renderers, interaction becomes **completely impossible** after selecting texture scaling, e.g., xBRZ x3.

I am attaching two savestates that skip lengthy intro for testing. The first one comes from the version with translation 0.5a applied, the second one from the original (due to GitHub policy, I had to pack them into a zip file).

[savestates.zip](https://github.com/user-attachments/files/24794728/savestates.zip)

**Comment by hrydgard** [comment]
@ghvxx43696 As mentioned in one of the other issues, using texture scaling and similar things that modify colors (other than post-processing) in this game is a bad idea - the game seems to depend on the exact color values. For some reason on iOS I guess the Vulkan backend becomes a little bit off. This is not seen on other platforms though - at least on Windows devices this works correctly...

**Comment by hrydgard** [comment]
Also, when interactions are difficult / impossible, does the triangle button (that shows targets) just do nothing? or does it show some, or all targets?

**Comment by ghvxx43696** [comment]
It shows all points of interest/targets and as far as I tested shows them in correct places. However, selecting object (if possible) can cause interaction with another person/object sometimes quite distant or located on “deeper” layer (tested on 4x resolution).

**Comment by hrydgard** [comment]
@ghvxx43696 Please sign up for TestFlight and try there. It should be working fine now in Vulkan.

https://testflight.apple.com/join/uNlhFG0m

**Comment by ghvxx43696** [comment]
After switching from OpenGL to Vulkan, the application crashes to the desktop immediately after selecting a game.

**Comment by hrydgard** [comment]
What? On iOS with the TestFlight version? I had absolutely no issues with vulkan on iOS when I tested recently.

**Comment by ghvxx43696** [comment]
Unfortunately, yes. Instant crash. On OpenGL, the game seems to have all the problems of the stable version. Additionally, multi touch does not work on screen buttons, so it is impossible to progress dialogues using fast-forward, which makes testing very tedious. However, the interface's support for vertical orientation and its minor improvements, such as icons, are very much appreciated.

**Comment by hrydgard** [comment]
What in the world? I'm not having any of those issues. I'm testing on an iPhone 12 Pro, what are you testing on?

There's just no way multi-touch isn't working unless you have changed some system setting on the device.

**Comment by ghvxx43696** [comment]
iPad 8th gen. iPadOS 17.5.1. I don't think it's a matter of settings. In fact, the beta version completely reset my custom button settings and restored the default layout. Maybe there are some differences between iPadOS and iOS? I don't know.

**Comment by hrydgard** [comment]
I've even ordered the game from Japan and I'm testing it myself on the iPhone. Vulkan works just perfectly. Multitouch works. I really don't understand how you're having all these problems...

Maybe it's a setting? Make sure that GPU texture upscaling isn't set to MMPX. Other than that I don't know, I'm trying every other graphical setting trying to make it crash, but no dice. Also do try to turn off any post processing shaders in "Display layout & effects".

**Comment by ghvxx43696** [comment]
At this point, I didn't care anymore and restored the app's default settings (and deleted game config). After restarting, everything works the same. Vulkan crashes, multi-touch doesn't work.

**Comment by hrydgard** [comment]
Never mind the last message, I saw now that you posted that it's an iPad.

**Comment by hrydgard** [comment]
But iPadOS 17? Why? I am not able to support older iOS versions than the latest, as there is no way to go back.

**Comment by ghvxx43696** [comment]
Umm... Its right in my previous post.

> iPad 8th gen. iPadOS 17.5.1

**Comment by hrydgard** [comment]
And also, it didn't reset your button layout. But there's now separate layouts for portrait and landscape, and portrait gets a fresh layout, so I can see why it has that effect.

Please upgrade to the latest iPadOS and try again.

**Comment by hrydgard** [comment]
In addition, I've now ordered an iPad 8th gen, seems I need one for testing.

I still have zero idea why multitouch could fail. It's just impossible.

**Comment by hrydgard** [comment]
OK, I've got my iPad 8th gen. Was hoping for it to come with an old iOS, but it was updated. Apparently it doesn't matter - it does indeed crash with Vulkan! Argh. Well, at least I can reproduce the problem. I'm getting some strange out-of-memory problem.

Maybe I'll start by updating MoltenVK. Anyway, there is hope.

**Comment by hrydgard** [comment]
I'm doing a huge bisect now from 1.19.3 to current master. Somewhere in between it broke. It seems correlated with a problem too where we are somehow not getting bilinear filtering on the UI shaders, making everything look a bit pixellated. Possibly this happened when I added support for portrait mode, but I don't know how that caused this problem....

**Comment by hrydgard** [comment]
No, it seems this broke with the previous MoltenVK upgrade. I'm going to have to downgrade.

- 4c8fb07e68981d3fe5ac3642234ab2126cf976f2

And in the long run, probably we should have a Metal backend, that'd be a significant effort though...

**Comment by hrydgard** [comment]
Alright, there will be a new TestFlight to try later this week. Hopefully the last one before release.

Thanks for reporting this, this turned out to be a really critical problem for these older devices. And you don't have to upgrade to iOS 26 if you don't want to - as it's not gonna help. The next TestFlight will work on your device though.

**Comment by hrydgard** [comment]
New TestFlight is up, Vulkan should work for you again.
